src/pkcs15init/profile.c

```diff
@@ -499,6 +499,7 @@ sc_profile_get_pin_info(struct sc_profile *profile,
 		int id, struct sc_pkcs15_auth_info *info)
 {
 	struct pin_info	*pi;
+	memset(info, 0, sizeof(*info));
```

Comment:
Posted to the previous version of the comments inline, as it still applies:

I think the memset here is more like patch hiding the other problem that the callers of `sc_profile_get_pin_info()` do not initialize the info argument and do not check the return values of this function, causing uninitialized reads.

My preference would be either or both of the following instead of blindly calling memset on a argument of the function:

* Initialize the info argument in all calls to this function
* Check return values of this function before using its results.

Reply:
ok, removed. I added some more initialization to fix other failures due to the missing memset, as you suggested.